src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,99 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_USER_LOGIN_ONE = "DummyUser";
+    private static final String DEFAULT_USER_LOGIN_TWO = "DummyUserTwo";
+    private static final String DEFAULT_USER_PASSWORD = "DummyUserPassword";
+    private static final String SHORT_LOGIN = "Login";
+    private static final String SHORT_PASSWORD = "pass";
+    private static final int DEFAULT_USER_AGE = 18;
+    private StorageDao storageDao = new StorageDaoImpl();
```

Comment:
Consider using `@BeforeAll` with a `static` initialization for `storageDao` if the instance doesn't need to be reset before each test. If the state of `storageDao` after each test is irrelevant to other tests, `@BeforeAll` could be more efficient.

Reply:
fix comment